frontend/src/requests/transaction.rs

```diff
@@ -1,19 +1,58 @@
+use serde_json::{json, Value};
+
 use crate::config::Config;
 
 use super::request;
 
-pub async fn ping(config: &Config) -> Result<String, String> {
+pub enum TransactionType {
+    Ping,
+    Pong,
+    PongAll,
+}
+
+pub async fn ping(config: &Config) -> Result<Value, Value> {
     let transaction_url = &config.transaction_url;
     let dest = &config.dest;
-    let full_endpoint = format!("http://{dest}{transaction_url}");
+    let endpoint = format!("http://{dest}{transaction_url}/ping");
+
+    let body = json!({
+        "sender": "erd1qyu5wthldzr8wx5c9ucg8kjagg0jfs53s8nr3zpz3hypefsdd8ssycr6th",
+        "value": 0.001
+    });
 
-    request::post_request("ping", &full_endpoint).await
+    request::post_request("ping", &endpoint, Some(&body)).await
 }
 
-pub async fn sc_setup(config: &Config) -> Result<String, String> {
+pub async fn sc_setup(config: &Config) -> Result<Value, Value> {
     let setup_url = &config.setup_url;
     let dest = &config.dest;
-    let full_endpoint = format!("http://{dest}{setup_url}");
+    let endpoint = format!("http://{dest}{setup_url}");
+
+    let body = json!(
```

Comment:
let's have a form, i think it's mentioned in a previous pr as well

Reply:
coming in next PR